packages/react-router/docs/guides/migrating.md

```diff
@@ -0,0 +1,143 @@
+
+```js
+// v3
+import routes from './routes'
+<Router history={browserHistory} routes={routes} />
+// or
+<Router history={browserHistory}>
+  <Route path='/' component={App}>
+    // ...
+  </Route>
+</Router>
+```
+
+With React Router v4, one of the big changes is that there are a number of different router components. Each one will create a `history` object for you. The `<BrowserRouter>` creates a browser history, the `<HashRouter>` creates a hash history, and the `<MemoryRouter>` creates a memory history.
+
+In v4, there is no centralized route configuration. Anywhere that you need to render content based on a route, you will just render a `<Route>` component.
+
+```js
+//v4
+<BrowserRouter>
+  <div>
+    <Route path='/about' component={About} />
+    <Route path='/contact' component={Contact} />
+  </div>
+</BrowserRouter>
+```
+
+One thing to note is that the router component must only be given one child element.
+
+```js
+// yes
+<BrowserRouter>
+  <div>
+    <Route path='/about' component={About} />
+    <Route path='/contact' component={Contact} />
+  </div>
+</BrowserRouter>
+
+// no
+<BrowserRouter>
+  <Route path='/about' component={About} />
+  <Route path='/contact' component={Contact} />
+</BrowserRouter>
+```
+
+## Routes
+
+In v3, the `<Route>` was not really a component. Instead, all of your application's `<Route>` elements were just used to created a route configuration object.
+
+```js
+/// in v3 the element
+<Route path='contact' component={Contact} />
+// was equivalent to
+{
+  path: 'contact',
+  component: Contact
+}
+```
+
+With v4, you layout your app's components just like a regular React application. Anywhere that you want to render content based on the location (specifically, its `pathname`), you render a `<Route>`.
+
+The v4 `<Route>` component is actually a component. so wherever you render a `<Route>` component, content will be rendered. When the `<Route>`'s `path` matches the current location, it will use its rendering prop (`component`, `render`, or `children`) to render. When the `<Route>`'s `path` does not match, it will render `null`.
+
+### Nesting Routes
+
+In v3, `<Route>`s were nested by passing them as the `children` of their parent `<Route>`.
+
+```js
+<Route path='parent' component={Parent}>
+  <Route path='child' component={Child} />
+  <Route path='other' component={Other} />
+</Route>
+```
+
+When a nested `<Route>` matched, React elements would be created using both the child and parent `<Route>`'s `component` prop. The child element would be passed to the parent element as its `children` prop.
+
+```js
+<Parent {...routeProps}>
+  <Child {...routeProps} />
+</Parent>
+```
+
+With v4, children `<Route>`s should just be rendered by the parent `<Route>`'s component.
+
+```js
+<Route path='parent' component={Parent} />
+
+const Parent = () => (
+  <div>
+    <Route path='child' component={Child} />
+    <Route path='other' component={Other} />
+  </div>
+)
+```
+
+### `on*` properties
```

Comment:
Some actual examples would be useful here. If this gets merged it will be a good place for someone to make a PR.